src/app/(main)/query-runner/_components/cql-autocompleter.ts

```diff
@@ -0,0 +1,372 @@
+import { getFullQueryAtCursor } from "@scylla-studio/app/(main)/query-runner/_components/utils";
+import { useCqlFilters } from "@scylla-studio/hooks/use-cql-filters";
+import { KeyspaceDefinition } from "@scylla-studio/lib/cql-parser/keyspace-parser";
+import { TableDefinition } from "@scylla-studio/lib/cql-parser/table-parser";
+import {
+  CompletionItem,
+  CompletionItemLabel,
+  editor,
+  languages,
+} from "monaco-editor";
+
+export type MonacoInstance = NonNullable<ReturnType<typeof useMonaco>>;
+
+// Define CQL keywords, functions, data types, and operators
+const keywords = [
+  "ADD",
+  "AGGREGATE",
+  "ALL",
+  "ALLOW",
+  "ALTER",
+  "AND",
+  "ANY",
+  "APPLY",
+  "AS",
+  "ASC",
+  "AUTHORIZE",
+  "BATCH",
+  "BEGIN",
+  "BY",
+  "CAST",
+  "COLUMNFAMILY",
+  "CREATE",
+  "CUSTOM",
+  "DELETE",
+  "DESC",
+  "DISTINCT",
+  "DROP",
+  "ENTRIES",
+  "EXISTS",
+  "FILTERING",
+  "FROM",
+  "FULL",
+  "FUNCTION",
+  "GRANT",
+  "IF",
+  "IN",
+  "INDEX",
+  "INFINITY",
+  "INSERT",
+  "INTO",
+  "IS",
+  "JSON",
+  "KEY",
+  "KEYSPACE",
+  "KEYSPACES",
+  "LANGUAGE",
+  "LIMIT",
+  "LOGIN",
+  "MATERIALIZED",
+  "MODIFY",
+  "NAN",
+  "NAMESPACE",
+  "NORECURSIVE",
+  "NOT",
+  "NULL",
+  "OF",
+  "ON",
+  "OPTIONS",
+  "OR",
+  "ORDER",
+  "PARTITION",
+  "PASSWORD",
+  "PER",
+  "PERMISSION",
+  "PERMISSIONS",
+  "PRIMARY",
+  "REVOKE",
+  "SCHEMA",
+  "SELECT",
+  "SET",
+  "SFUNCTION",
+  "STATIC",
+  "STORAGE",
+  "SUPERUSER",
+  "TABLE",
+  "TOKEN",
+  "TRUNCATE",
+  "TTL",
+  "TYPE",
+  "UNLOGGED",
+  "UPDATE",
+  "USE",
+  "USER",
+  "USERS",
+  "USING",
+  "VALUES",
+  "VIEW",
+  "WHERE",
+  "WITH",
+  "WRITETIME",
+];
+
+const functions = [
+  "blobAsBigint",
+  "blobAsBoolean",
+  "blobAsDecimal",
+  "blobAsDouble",
+  "blobAsFloat",
+  "blobAsInt",
+  "blobAsText",
+  "blobAsTimestamp",
+  "blobAsUUID",
+  "blobAsVarchar",
+  "blobAsVarint",
+  "dateOf",
+  "now",
+  "unixTimestampOf",
+  "uuid",
+  "timeuuid",
+  "minTimeuuid",
+  "maxTimeuuid",
+  "toDate",
+  "toTimestamp",
+  "toUnixTimestamp",
+  "writetime",
+  "ttl",
+  "token",
+  "bigintAsBlob",
+  "booleanAsBlob",
+  "dateAsBlob",
+  "decimalAsBlob",
+  "doubleAsBlob",
+  "floatAsBlob",
+  "inetAsBlob",
+  "intAsBlob",
+  "textAsBlob",
+  "timestampAsBlob",
+  "timeuuidAsBlob",
+  "uuidAsBlob",
+  "varcharAsBlob",
+  "varintAsBlob",
+];
+
+const dataTypes = [
+  "ascii",
+  "bigint",
+  "blob",
+  "boolean",
+  "counter",
+  "date",
+  "decimal",
+  "double",
+  "float",
+  "frozen",
+  "inet",
+  "int",
+  "list",
+  "map",
+  "set",
+  "smallint",
+  "text",
+  "time",
+  "timestamp",
+  "timeuuid",
+  "tinyint",
+  "tuple",
+  "uuid",
+  "varchar",
+  "varint",
+];
+
+const operators = [
+  "=",
+  ">",
+  "<",
+  ">=",
+  "<=",
+  "!=",
+  "+",
+  "-",
+  "*",
+  "/",
+  "%",
+  "IN",
+  "CONTAINS",
+  "CONTAINS KEY",
+  "AND",
+  "OR",
+  "NOT",
+  "IS",
+  "NULL",
+  "IS NOT NULL",
+  "IS NULL",
+];
```

Comment:
pq não move essas constantes para um arquivo separado para facilitar a leitura?

Reply:
Yo soy burrito mi cabeça no hablas frontend

Reply:
At some point, the file `cql-autocompleter.ts` was created to be this abstraction. 

I think we can keep these constants in the same file for now, since we don't know how far the scope of the implementation will grow.

See #48